gobblin-scheduler/src/main/java/gobblin/scheduler/JobScheduler.java

```diff
@@ -343,45 +350,50 @@ public void onFileCreate(File file) {
        */
       @Override
       public void onFileChange(File file) {
```

Comment:
Can you put some comments here for the situation when the changes of two files (e.g, one is ".properties", and one is ".job") are detected in the same time window by FileAlterationListener? The override order may be confusing.